source/app/models/url.rb

```diff
@@ -0,0 +1,22 @@
+class Url < ActiveRecord::Base
+  before_save :shorten
+
+  validates :shortened, uniqueness: true
+  validate :protocol_check, :resolvable_check
+
+  def shorten
+    self.shortened ||= SecureRandom.urlsafe_base64(5)
+  end
+
+  def protocol_check
+    errors[:original] <<  "URL must begin with http:// or https://" unless original.match(/^https?:\/\//)
+  end
+
+  def resolvable_check
+    begin
+      HTTParty.get(original).code == 200
```

Comment:
Yeah and validation too. Let's talk about it at 4p. A couple of people made the same decision. Within the scope of this challenge, its sufficient but there's a bigger lesson about handling user input here.